vllm/v1/core/sched/scheduler.py

```diff
@@ -568,7 +570,8 @@ def update_from_output(
 
         new_running: list[Request] = []
         outputs: list[EngineCoreOutput] = []
-        spec_decoding_stats: Optional[SpecDecodingStats] = None
+        # spec_decoding_stats: Optional[SpecDecodingStats] = None
+        spec_decoding_stats = self.spec_decoding_stats
```

Comment:
cache the `spec_decoding_stats` so that it keeps a running metric instead of reinit it every engine step